playground/src/file-upload/file-upload.js

```diff
@@ -0,0 +1,85 @@
+// Copyright 2020 The AMPHTML Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import './file-upload.scss';
+import template from './file-upload.hbs';
+
+import events from '../events/events.js';
+import Dropzone from 'dropzone';
+import * as EmailLoader from '../email-loader/email-loader.js';
+
+export const EVENT_FILE_UPLOADED = 'event-file-uploaded';
+
+const ALLOWED_FILE_TYPE = 'message/rfc822';
+
+export default function createFileUpload(target) {
+  return new FileUpload(target);
+}
+
+class FileUpload {
```

Comment:
This whole class isn't as generic as it seems - especially in comparison to the input bar. It binds to `EmailLoader.*` events, always has an envelope as an icon, only accepts `message/rfc822`, ...